2025-04-02 12:48:23,167 INFO: Microblog startup [in C:\Users\Ethan\PycharmProjects\Travel Buddy\app\__init__.py:28]
2025-04-02 12:48:26,082 ERROR: Exception on / [GET] [in C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 284, in decorated_view
    elif not current_user.is_authenticated:
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\werkzeug\local.py", line 318, in __get__
    obj = instance._get_current_object()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\werkzeug\local.py", line 526, in _get_current_object
    return get_name(local())
                    ^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 25, in <lambda>
    current_user = LocalProxy(lambda: _get_user())
                                      ^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 370, in _get_user
    current_app.login_manager._load_user()
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\login_manager.py", line 347, in _load_user
    raise Exception(
Exception: Missing user_loader or request_loader. Refer to http://flask-login.readthedocs.io/#how-it-works for more info.
2025-04-02 12:51:01,564 INFO: Microblog startup [in C:\Users\Ethan\PycharmProjects\Travel Buddy\app\__init__.py:28]
2025-04-02 12:51:04,872 ERROR: Exception on / [GET] [in C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 284, in decorated_view
    elif not current_user.is_authenticated:
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\werkzeug\local.py", line 318, in __get__
    obj = instance._get_current_object()
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\werkzeug\local.py", line 526, in _get_current_object
    return get_name(local())
                    ^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 25, in <lambda>
    current_user = LocalProxy(lambda: _get_user())
                                      ^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 370, in _get_user
    current_app.login_manager._load_user()
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\login_manager.py", line 347, in _load_user
    raise Exception(
Exception: Missing user_loader or request_loader. Refer to http://flask-login.readthedocs.io/#how-it-works for more info.
2025-04-02 12:51:44,385 INFO: Microblog startup [in C:\Users\Ethan\PycharmProjects\Travel Buddy\app\__init__.py:28]
2025-04-02 12:51:45,734 ERROR: Exception on / [GET] [in C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py:875]
Traceback (most recent call last):
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\app\routes.py", line 15, in index
    return render_template('index.html', title='Home')
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\templating.py", line 127, in _render
    app.update_template_context(context)
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask\app.py", line 530, in update_template_context
    context.update(self.ensure_sync(func)())
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 405, in _user_context_processor
    return dict(current_user=_get_user())
                             ^^^^^^^^^^^
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\utils.py", line 370, in _get_user
    current_app.login_manager._load_user()
  File "C:\Users\Ethan\PycharmProjects\Travel Buddy\.venv\Lib\site-packages\flask_login\login_manager.py", line 347, in _load_user
    raise Exception(
Exception: Missing user_loader or request_loader. Refer to http://flask-login.readthedocs.io/#how-it-works for more info.
2025-04-03 15:59:48,823 INFO: Microblog startup [in C:\Users\Ethan\PycharmProjects\Travel Buddy\app\__init__.py:28]
